### PR TITLE
Introduce GCS, IAM support for Loki and Remote state management

### PR DESCRIPTION
**Pull request by amar-opsverse**

### CODE REVIEW

File: gcp/opsverse-gke/vars.tfvars [hunk 1]
```diff
@@ -1,10 +1,20 @@
...
+cluster_name   = "opsverse-gke-cluster"
+# The bucket name should be of the format devopsnow-loki-<customer_name>-<stack_name>
+#If customer name is kylaq , the stack name is decided as alpha then bucket_name = "devopsnow-loki-kylaq-alpha" 
+bucket_name = "devopsnow-loki-<customer_name>-<stack_name>"
+
+
+#The customer_stack should be of the format <customer_name>-<stack_name> 
+#If customer name is kylaq , the stack name is decided as alpha then customer_stack = "kylaq-alpha"
+#Please Note: The stack name used here should be the same one when trying to launch the stack through the admin console.
+customer_stack = "<customer_name>-<stack_name>"
```
Comment (aravind-opsverse) [inline gcp/opsverse-gke/vars.tfvars]: Why is this being called `customer_stack`? This feels ambiguous for the customers. Can this naming convention be simplified?

File: gcp/opsverse-gke/vars.tfvars [hunk 2]
```diff
@@ -17,97 +27,97 @@ node_locations      = "us-east4-a,us-east4-b,us-east4-c"
 # 3 Public Subnets and 3 Private Subnets by default
 # This can be changed as per requirements
 subnets = [
+  {
+    subnet_name   = "opsverse-gke-cluster-apps-public-a"
```
Comment (aravind-opsverse) [inline gcp/opsverse-gke/vars.tfvars:30]: What is the change here? It becomes hard for reviewers to understand the change/diff. 
Please avoid mixing the lint changes with functional changes going forward.

Comment (amar-opsverse) [reply gcp/opsverse-gke/vars.tfvars:30]: There are no functional changes here, only **terraform fmt** was applied to format the .tf files according to standard guidelines .In the future, I'll make sure to distinguish formatting modifications from functional upgrades.

File: gcp/opsverse-gke/vars.tfvars (hunk 1 above)
Comment (amar-opsverse) [reply gcp/opsverse-gke/vars.tfvars]: In our org. we follow two different conventions for service accounts created in GCP and Kubernetes, so to address this i used this customer_stack input that binds the gcp sa to the k8s sa. I tried to keep the variable simple while including some logic, so I used customer_stack as customername-_stackname, If you have any better suggestions i am open to it

File: gcp/opsverse-gke/vars.tfvars [hunk 3]
```diff
@@ -1,10 +1,20 @@
...
-cluster_name        = "opsverse-gke-cluster"
+gcp_project_id = "opsverse-dev-poc"
+gcp_region     = "us-east4"
+cluster_name   = "opsverse-gke-cluster"
+# The bucket name should be of the format devopsnow-loki-<customer_name>-<stack_name>
+#If customer name is kylaq , the stack name is decided as alpha then bucket_name = "devopsnow-loki-kylaq-alpha" 
+bucket_name = "devopsnow-loki-<customer_name>-<stack_name>"
+
+
+#The customer_stack should be of the format <customer_name>-<stack_name>
```
Comment (aravind-opsverse) [inline gcp/opsverse-gke/vars.tfvars]: All this information has to be added to `README.md`. Customers refer Readme file. Let's add all these details.

File: gcp/opsverse-gke/vars.tfvars [hunk 4]
```diff
@@ -1,10 +1,20 @@
-gcp_project_id      = "opsverse-dev-poc"
-gcp_region          = "us-east4"
-cluster_name        = "opsverse-gke-cluster"
+gcp_project_id = "opsverse-dev-poc"
+gcp_region     = "us-east4"
+cluster_name   = "opsverse-gke-cluster"
+# The bucket name should be of the format devopsnow-loki-<customer_name>-<stack_name>
```
Comment (aravind-opsverse) [inline gcp/opsverse-gke/vars.tfvars:4]: All this information has to be added to `README.md`. Customers refer Readme file. Let's add all these details.

Comment (aravind-opsverse) [reply gcp/opsverse-gke/vars.tfvars:4]: Let's highlight all this information in the README using `Important` tag.

File: gcp/opsverse-gke/vars.tfvars (hunk 1 above)
Comment (amar-opsverse) [reply gcp/opsverse-gke/vars.tfvars]: Done

File: gcp/opsverse-gke/vars.tfvars (hunk 3 above)
Comment (amar-opsverse) [reply gcp/opsverse-gke/vars.tfvars]: Done

File: gcp/opsverse-gke/vars.tfvars (hunk 4 above)
Comment (amar-opsverse) [reply gcp/opsverse-gke/vars.tfvars:4]: Done

**Review by aravind-opsverse — APPROVED** [review]
LGTM

**Pull request merged**